src/main/java/io/nats/client/support/IncomingHeadersProcessor.java

```diff
@@ -39,14 +39,20 @@ public IncomingHeadersProcessor(byte[] serialized) {
 
         // does the header end properly
         serializedLength = serialized.length;
-        new Token(serialized, serializedLength, serializedLength - 2, TokenType.CRLF);
+        Token terminus = new Token(serialized, serializedLength, serializedLength - 2, TokenType.CRLF);
```

Comment:
Changes to this class are to support having both a status and headers in an HMSG. Before if it found a status, it just stopped.